### PR TITLE
west: Update liblc3

### LINKED ISSUE #10: Port as external project
It will be better to port this codebase as external project (https://docs.zephyrproject.org/latest/develop/modules.html)
The root codebase has landed to github (https://github.com/google/liblc3)
The last version includes many speed improvements, especially for ARM Cortex M series.

### PR DESCRIPTION
**Pull request by MariuszSkamra**
Start using new liblc3 fork.

Fixes: https://github.com/zephyrproject-rtos/liblc3codec/issues/10
Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

**Comment by MariuszSkamra** [comment]
@asoulier FYI

**Comment by Thalley** [comment]
@MariuszSkamra 

> As the LC3 code has been published on github, the library can be ported as external project, so that there is no need to copy the code.
> 
> Relates: [google/liblc3#7](https://github.com/google/liblc3/pull/7) Fixes: [zephyrproject-rtos/liblc3codec#10](https://github.com/zephyrproject-rtos/liblc3codec/issues/10) Signed-off-by: Mariusz Skamra [mariusz.skamra@codecoup.pl](mailto:mariusz.skamra@codecoup.pl)

What about the existing (copied) code?

**Comment by MariuszSkamra** [comment]
> What about the existing (copied) code?

@Thalley I guess it's not needed anymore to maintain. But I don't think I have permission to archive/delete it... @nashif ^^

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM
Did not test it.

**Comment by MariuszSkamra** [comment]
> LGTM Did not test it.

Me neither... Just kidding, here's some music I recorded streaming from Pixel 6
[test.zip](https://github.com/zephyrproject-rtos/zephyr/files/9012223/test.zip)

**Review by Thalley — COMMENTED** [review]
LGTM too, but I do not have the necessary insight and understanding of how these modules work to be confident to give an approval (and then there's the question about whether the existing code for the module should be removed as well).

**Comment by mmahadevan108** [comment]
I think this repo should be cloned into the `zephyrproject-rtos` space per the below requirement:
https://docs.zephyrproject.org/latest/develop/modules.html#module-repositories

**Comment by Casper-Bonde-Bose** [comment]
> > LGTM Did not test it.
> 
> Me neither... Just kidding, here's some music I recorded streaming from Pixel 6 [test.zip](https://github.com/zephyrproject-rtos/zephyr/files/9012223/test.zip)

Nice! - it seems like there is an issue with either the play-back (or the PLC) I do see for example a 22ms gap at 12.288 in the test.wav file you shared. But that is unrelated to this PR :-)

**Comment by Casper-Bonde-Bose** [comment]
> I think this repo should be cloned into the zephyrproject-rtos space per the below requirement:
@mmahadevan108 I was of the impression that this was before all build-related files could be places in the Zephyr code base, but I see that the guide still suggests to clone the external code - hence I must be wrong.

Google created the external repo to allow Zephyr and other open source projects (like BlueZ) to use a shared code-base to do joint work on optimizations. Hence everyone will benefit from keeping it external - as it is the hope that Zephyr developers will contribute to the LC3 codec as well. If we keep it external it is straight forward for everyone to do contributions, but if we clone you need to know quite a lot of git and west magic to figure out how you setup zephyr to use the external repo while testing your contribution(as Zephyr changes should not go into cloned repositories).

@carlescufi would you happen to know if it is OK to use an external repo, or we should clone it. And if we clone it, what would be the steps to replace the current code-base with one that includes the history of the external repository?

If there is agreement to clone the code, feel free to make a PR that replaces all content of zephyrproject-rtos/liblc3codec (it is just an older copy of the same code from before it was moved into a stand alone git-project.)

**Review by carlescufi — DISMISSED** [review]
This is not allowed

**Comment by Thalley** [comment]
I am a bit out of loop with this one: 

What's the plan here, and are we now getting the LC3 directly from the other Github repository, or what is going on?

File: modules/liblc3codec/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 if(CONFIG_LIBLC3CODEC)
 
 zephyr_library_named(liblc3codec)
-zephyr_library_compile_options(-O3 -ffast-math -Wno-array-bounds)
+zephyr_library_compile_options(-O3 -std=c11 -ffast-math -Wno-array-bounds)
```
Comment (Casper-Bonde-Bose) [inline modules/liblc3codec/CMakeLists.txt:4]: Why did you add this? I don't think it is required?

**Review by Casper-Bonde-Bose — APPROVED** [review]
LGTM - a single question - nothing that should block this commit.
I tested basically the same - just without the c11 bit.

**Comment by MariuszSkamra** [comment]
> LGTM - a single question - nothing that should block this commit. I tested basically the same - just without the c11 bit.

I recall I had some problems compiling the code without `c11`. Maybe it's the matter of GCC version...

**Comment by Casper-Bonde-Bose** [comment]
> I am a bit out of loop with this one:
> 
> What's the plan here, and are we now getting the LC3 directly from the other Github repository, or what is going on?

Yes almost. Google moved the LC3 code to a separate repo that is now easy to clone in the zephyr github space. So https://github.com/zephyrproject-rtos/liblc3 is a clone of this external Google repo, and this PR changes from liblc3codec (which only have a copy of the LC3 source-files) to liblc3 which is a clone with history of the same library, but is far easier to maintain.
IMO liblc3codec should be deleted as soon as possible as liblc3 is the same thing, just with git history and a few fixes - no API changes.

File: modules/liblc3codec/CMakeLists.txt (hunk 1 above)
Comment (MariuszSkamra) [reply modules/liblc3codec/CMakeLists.txt:4]: C11 added `alignas` specifier https://en.cppreference.com/w/cpp/language/alignas used in liblc3codec. Without the `c11` option I get compilation errors:

```
/home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf_arm.h: In function 'arm_resample_8k_12k8':
/home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf_arm.h:124:34: error: expected expression before 'int32_t'
  124 |     static const int16_t alignas(int32_t) h[2*8*12] = {
      |                                  ^~~~~~~
/home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf_arm.h:124:26: error: expected identifier or '(' before ')' token
  124 |     static const int16_t alignas(int32_t) h[2*8*12] = {
      |                          ^~~~~~~
In file included from /home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf.c:23:
/home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf_arm.h:144:31: error: 'h' undeclared (first use in this function)
  144 |         8, (const int16x2_t *)h, hp50, (int16x2_t *)x, y, n);
      |                               ^
/home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf_arm.h:144:31: note: each undeclared identifier is reported only once for each function it appears in
In file included from /home/mariuszskamra/zephyr-sdk/arm-zephyr-eabi/arm-zephyr-eabi/sys-include/math.h:6,
                 from /home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/fastmath.h:27,
                 from /home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/common.h:27,
                 from /home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf.h:29,
                 from /home/mariuszskamra/zephyrproject/modules/lib/liblc3codec/src/ltpf.c:19:
```

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
